src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,122 @@
+
+With everything hopefully set up properly, you may now run *rosrun topic_training Demo* or *rosrun topic_training Demo.py*. In the terminal, you should see the program continually print "hello world {count} I heard: [hello world {count}]" until the terminal receives Ctrl-C or the program crashes.
+
+### rqt_graph and rqt_plot
+In another terminal window, run *rosrun rqt_graph rqt_graph*. This will show a graph representation of the current topic you are running. The name of the file (+ the ID number if you are using Python) should be encircled with an arrow named after the topic (in this case "chatter") pointing to itself since the file contains both the Publisher and Subscriber.
+
+There is also another command that helps you visualize the nodes and topics. Type Ctrl-C to stop running Demo(.cpp or .py) and, instead, run *rosrun topic_training PlotDemo* or *rosrun topic_training PlotDemo.py*. You should not see anything print. The file PlotDemo(.cpp or .py) is a Publisher (no Subscriber) that sends sine values to the same topic ("chatter"). If you run *rosrun rqt_plot rqt_plot*, you should see a sine graph being plotted onto the screen. The command rqt_plot is only able to be used on topics with a numerical type. Using it with the file Demo(.cpp or .py) would not have worked. 
+
+If you run rqt_graph on the PlotDemo(.cpp or .py) file, it should show just the name of the file (+ the ID number if you are using Python) by itself with no arrow. If you run rqt_graph when both files are running, there might be an error in the terminal, but the graph application should show both files. PlotDemo(.cpp or .py) will be by itself with no arrow, and Demo(.cpp or .py) will be by itself with or without an arrow depending on the order you executed the two files.
+
+Feel free to look through the file PlotDemo(.cpp or .py) to see how the math works and to see another example of a Publisher but with a different message type. For simplicity's sake, the comments in these files are only explaining the math since you should already understand the basic ROS functions from the previous section of this training.
+
+### Defining a Custom Topic Type
+A topic's type is determined by the type of the message being published. If you run the command *rostopic type chatter*, you should see the terminal print out "std_msgs/String" while roscore and rosrun are executing Demo(.cpp or .py). The command will print out "std_msgs/Float64" if you are executing PlotDemo(.cpp or .py).
+
+A topic can have messages of type int8, int16, int32, int64, uint*, float32, float64, String, time, duration, other msg files, and array[] of variable length or fixed length. To message a different type, just change where ever there would be a String in Demo(.cpp or .py) to your desired type and update the includes (C++) or imports (Python) at the top of the file.
```

Comment:
Could add a short sentence or paragraph at the end of this segment to remind students to use message types instead of variable types. For instance using `std_msgs/String` instead of `String`.

Reply:
Fixed in a8f38fc

Reply:
Fixed in a8f38fc.